Redmine#4309: document body inheritance via inherit_from

tzz opened the pull request:
see https://github.com/cfengine/core/pull/2320#issuecomment-141394427

nickanderson commented on guide/language-concepts/bodies.markdown:
```diff
@@ -55,6 +55,57 @@ The attributes within the body are then type specific. Bodies of type `perms` co
+
+    body perms system_inherited
+    {
+      inherit_from => mog("644", "root", "root");
```
What if it has inherit_from => mog("644", "root", "root"); and then mode => 600 specified within the system_inherited body?

tzz replied:
Latest in the inheritance chain wins.  I didn't think of that in the example, though.  Do you think it's worth including it (keeping in mind it's a beginner document) or should I put it elsewhere?

nickanderson replied:
It just seems like the kind of question someone would ask. I would add an example for it, just to show that you can inherit attributes that are set in other bodies, and that you can override them.

nickanderson commented on guide/language-concepts/bodies.markdown:
```diff
@@ -55,6 +55,57 @@ The attributes within the body are then type specific. Bodies of type `perms` co
+#### Body Inheritance
+
+CFEngine 3.8 introduced body inheritance via the `inherit_from`
+attribute. It's a parameterized single-inheritance system, so a body can only inherit from one other body, and it can apply parameters.
```
You say single-inheritance, but that might confuse someone so that they think you cannot inherit from a body that inherits from another body.

```cf3
body perms mode_system
{
  mode => "644";
}
body perms system
{
  inherit_from => mode_system;
  owner => "root";
  group => "root"
}
body perms system_inherited
{
  inherit_from => system;
}
```

nickanderson commented:
There should be documentation for the body attributes. Unfortunately I don't see that we have any way of easily documenting common body attributes (only bodies that are common to all promise types). So each attribute body will need its own blurb. I think that blurb should link back to the section in the language concepts.

tzz commented:
OK, I elaborated in the language concepts and added a new section to the common promise attributes.  I think that's a good place for it since there's only two of these.

I also added docs about the universal `meta` body attribute since I don't see docs for https://github.com/cfengine/core/pull/2088 anywhere.

nickanderson commented on guide/language-concepts/bodies.markdown:
````diff
@@ -55,6 +55,97 @@ The attributes within the body are then type specific. Bodies of type `perms` co
+    }
+
+```
+
````
Great!

nickanderson commented on reference/promise-types.markdown:
```diff
@@ -34,6 +34,82 @@ depends on the [bundle][bundles] type:
 See each promise type's reference documentation for detailed lists of available
 attributes.
 
+## Common Body Attributes
```
Thanks!
